### ISSUE TITLE
Provide checksums for your releases 

### ISSUE THREAD

**Issue by clmcgrath** [coreybutler/nvm-windows#256]
Please provide some way to obtain a checksum for your file release downloads, 
our package management system requires these in to install all external packages 

### My issue is related to (check only those which apply):

sha256 or md5 checksums

### Expected Behavior

Check sum provided for a release 

### Actual Behavior

no checksum available for a release

**Comment by ozbarshalom** [comment]
#257 
I see that you added a checksum generator, does this checksum being checked?

**Comment by clmcgrath** [comment]
Intention is to add to release for verifying with nuget and package systems or what not 
Is generated using the chocolatey checksum tool

**Comment by ozbarshalom** [comment]
Do you mean verify checksum on each release download?
after [downloading ](https://github.com/coreybutler/nvm-windows/blob/master/src/nvm/web/web.go#L55)release, check checksum ?
It can be done with [md5 go package](https://golang.org/pkg/crypto/md5/)

**Comment by clmcgrath** [comment]
Currently using sha256 check sums of that tool supports that then that
would work fine

**Comment by clmcgrath** [comment]
the intention is to provide some sort compatibility for  chocolatey packaging ,  as this currently requires checksum validation unless the user explicitly disables this functionality ,  i can go without it but package adoption is likely to be much lower,  and will likely not be accepted to the public feed

**Comment by clmcgrath** [comment]
the most recent release and future releases are the only ones that really would require this

**Comment by coreybutler** [comment]
So the checksum generator submitted in a PR isn't actually working when I build the executables and I haven't had time to troubleshoot it. I believe there are some assumed dependencies in the checksum generator that silently fail if they don't exist... but that's just my working theory. This is on my to-do list... I'm just pressed for time with the upcoming launch of Fenix.

**Comment by clmcgrath** [comment]
It's part of chocolatey
I may be able to submit it as a standalone exe if you need me to

**Comment by clmcgrath** [comment]
It's the one used for check sum validation in chocolatey packages as well

**Comment by coreybutler** [comment]
I've utilized certutil to generate md5 checksums in https://github.com/coreybutler/nvm-windows/releases/tag/1.1.7

**Comment by fazl** [comment]
tl;dr Please consider offering stronger checksums than SHA-1.  (MD5 cannot be trusted.)

I downloaded the zip file and was surprised the checksum didn't match any that were offered by 7-zip.. till I looked carefully and noticed you're giving MD5 checksums.

Even the successor to MD5 (SHA-1) was considered insecure in 2017.  From https://nakedsecurity.sophos.com/2017/02/23/bang-sha-1-collides-at-38762cf7f55934b34d179ae6a4c80cadccbb7f0a/ :

"For many years [...] MD5 was widely used [...] but it is now forbidden in the cryptographic world because [...] MD5 collisions are easy to generate on purpose, so the algorithm can no longer be trusted."
